### ISSUE TITLE
Tests are not isolated from the environment

### ISSUE THREAD

**Issue by IanManske** [nushell/nushell#12958]
### Describe the bug

Processes/`Command`s spawned in tests currently inherit the environment variables of the command running the tests. This can cause tests to fail in certain cases, depending on what environment variables happen to be present when the tests are started.

### How to reproduce

On current main, `cargo test --workspace` succeeds. But, `cargo nextest run --workspace` fails since it adds a `NEXTEST` environment variable. This causes the `hides_environment_from_child` test to fail, since the error message changes from the expected `cannot find column` to a `did you mean` error.

Similarly, depending on the locale, some tests will fail #10018.
See also #8463.

### Expected behavior

Tests should be isolated from the environment.

We should be using `Command::clear_env` in `nu_run_test` and other areas. However, Windows needs some environment variables to be preserved, as otherwise processes will fail to spawn. See related #9976.

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.93.1                                          |
| major              | 0                                               |
| minor              | 93                                              |
| patch              | 1                                               |
| branch             | main                                            |
| commit_hash        | 6c649809d38a687cd63b59ab028b7f30207e5663        |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)             |
| rust_channel       | 1.77.2-x86_64-unknown-linux-gnu                 |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)             |
| build_time         | 2024-05-22 22:48:00 -04:00                      |
| build_rust_channel | release                                         |
| allocator          | mimalloc                                        |
| features           | default, sqlite, system-clipboard, trash, which |
| installed_plugins  |                                                 |

**Comment by fdncred** [comment]
maybe i've misunderstood. do you mean the tests aren't passing or cargo nextest doesn't run at all? if it's about tests passing then i haven't finished running them all yet.

**Comment by IanManske** [comment]
Oof, it looks like a different test is failing on Windows and so it didn't get to the other test.

> Just to be thorough, here's what the output looks like on Ubuntu WSL.

Yep, that's what I'm seeing.

**Comment by cptpiepmatz** [comment]
Test using #17733 do not inherit environment from their parent.
